### PR TITLE
Document news API endpoints

### PR DESCRIPTION
**Pull request by cl8n**
supersedes #7246 , my mistake that I didn't see it before starting work on this, but it does seem abandoned and incomplete anyway.

also removes the now unused `public` scope from these routes

### CODE REVIEW

File: app/Http/Controllers/NewsController.php [hunk 1]
```diff
@@ -8,15 +8,79 @@
...
+     * news_sidebar.current_year | number                  | Year of the first post's publish time, or current year if no posts returned.
+     * news_sidebar.news_posts   | [NewsPost](#newspost)[] | All posts published during `current_year`.
+     * news_sidebar.years        | number[]                | All years during which posts have been published.
+     * search.limit              | number                  | Clamped limit input.
+     * search.sort               | string                  | Always `published_desc`.
+     * cursor                    | [Cursor](#cursor)       | |
+     *
+     * <aside class="notice">
+     *   <a href="#newspost">NewsPost</a> collections queried by year will also include posts published in November and December of the previous year if the current date is the same year and before April.
+     * </aside>
```
Comment (cl8n) [inline app/Http/Controllers/NewsController.php:37]: this is a lot to make sense of for a one-sentence notice but idk how else to word it 🤔

File: app/Http/Controllers/NewsController.php [hunk 2]
```diff
@@ -52,6 +116,53 @@ public function index()
...
+     *
+     * Returns details of the specified news post.
+     *
+     * ---
+     *
+     * ### Response Format
+     *
+     * Returns a [NewsPost](#newspost) with `content` and `navigation` included. Querying by ID will return a redirect to query by slug.
+     *
+     * @urlParam news string News post slug or ID. Example: 2021-04-28-new-featured-artist-emilles-moonlight-serenade
```
Comment (nanaya) [inline app/Http/Controllers/NewsController.php:130]: missing `required`

File: app/Http/Controllers/NewsController.php [hunk 3]
```diff
@@ -52,6 +116,53 @@ public function index()
...
+    /**
+     * Get News Post
+     *
+     * Returns details of the specified news post.
+     *
+     * ---
+     *
+     * ### Response Format
+     *
+     * Returns a [NewsPost](#newspost) with `content` and `navigation` included. Querying by ID will return a redirect to query by slug.
```
Comment (nanaya) [inline app/Http/Controllers/NewsController.php:128]: The redirect should probably be skipped for api request instead to match user api behavior

File: resources/views/docs/_structures/news_post.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+Field        | Type                    | Description
+-------------|-------------------------|------------
+id           | number                  | |
+author       | string                  | |
+edit_url     | string                  | Link to the file view on GitHub.
+first_image  | string?                 | Link to the first image in the document.
+published_at | [Timestamp](#timestamp) | |
+updated_at   | [Timestamp](#timestamp) | |
+slug         | string                  | Unique identifier made by joining the publish date and title.
+title        | string                  | |
```
Comment (nanaya) [inline resources/views/docs/_structures/news_post.md:12]: sort (also for others)

File: resources/views/docs/_structures/news_post.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+Field        | Type                    | Description
+-------------|-------------------------|------------
+id           | number                  | |
+author       | string                  | |
+edit_url     | string                  | Link to the file view on GitHub.
+first_image  | string?                 | Link to the first image in the document.
+published_at | [Timestamp](#timestamp) | |
+updated_at   | [Timestamp](#timestamp) | |
+slug         | string                  | Unique identifier made by joining the publish date and title.
```
Comment (nanaya) [inline resources/views/docs/_structures/news_post.md:11]: The relation to publish date and title isn't actually in the code. It's just the filename used in github

**Review by nanaya — CHANGES_REQUESTED** [review]

**Review by nanaya — APPROVED** [review]

**Pull request merged**